In [5]:
import pandas as pd
data = {
 "calories": [420, 380, 390],
 "duration": [50, 40, 45]
}
df = pd.DataFrame(data)
df

calories  duration
0       420        50
1       380        40
2       390        45

In [6]:
df["calories"] # เอาแค่ cloumn ที่เลือกในที่นี้เลือก column "calories"

0    420
1    380
2    390
Name: calories, dtype: int64

In [7]:
df.loc[0] # เอาแถวที่เลือกในที่นี้เลือก แถวที่ 0

calories    420
duration     50
Name: 0, dtype: int64

pd.read_csv('data.csv')


In [13]:
import pandas as pd
df = pd.read_csv('data.csv')

In [ ]:
df.head()

Duration  Pulse  Maxpulse  Calories
0        60    110       130     409.1
1        60    117       145     479.0
2        60    103       135     340.0
3        45    109       175     282.4
4        45    117       148     406.0

In [ ]:
df.tail()

Duration  Pulse  Maxpulse  Calories
164        60    105       140     290.8
165        60    110       145     300.0
166        60    115       145     310.2
167        75    120       150     320.4
168        75    125       150     330.4

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169 entries, 0 to 168
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Duration  169 non-null    int64  
 1   Pulse     169 non-null    int64  
 2   Maxpulse  169 non-null    int64  
 3   Calories  164 non-null    float64
dtypes: float64(1), int64(3)
memory usage: 5.4 KB


In [10]:
df.fillna(0, # แทน na เป็น 0
          inplace=True) # เปลี่ยนค่า df

In [8]:
df.dropna(inplace=True) # เปลี่ยนค่า df


In [14]:
print(df.to_string()) # printทุกค่า

     Duration  Pulse  Maxpulse  Calories
0          60    110       130     409.1
1          60    117       145     479.0
2          60    103       135     340.0
3          45    109       175     282.4
4          45    117       148     406.0
5          60    102       127     300.0
6          60    110       136     374.0
7          45    104       134     253.3
8          30    109       133     195.1
9          60     98       124     269.0
10         60    103       147     329.3
11         60    100       120     250.7
12         60    106       128     345.3
13         60    104       132     379.3
14         60     98       123     275.0
15         60     98       120     215.2
16         60    100       120     300.0
17         45     90       112       NaN
18         60    103       123     323.0
19         45     97       125     243.0
20         60    108       131     364.2
21         45    100       119     282.0
22         60    130       101     300.0
23         45   

โจทย์
1. sum Duration
2. print Duration ทุกอันออกมา โดยไม่เอาเลขที่มากกว่า 120
3. print ทุก row ออกมา โดยไม่เอา Pulse ที่น้อยกว่า 100
4. print ทุก row ที่ Calories มากกว่า Pulse
5. print ทุก row ที่ผลลัพธ์ของ Maxpulse ลบ Pulse ได้มากกว่า 10